src/dictionary/dictionary.h

```diff
@@ -372,6 +372,25 @@ dictionary_find(
 	ion_dict_cursor_t	**cursor
 );
 
+/**
+@brief		Tests the supplied @p key against the predicate registered in the
```

Comment:
This documentation for `test_predicate` isn't the best (but I understand you weren't the original author of them). Can we expand some more detail here?